src/lib/DpApi.js

```diff
@@ -39,28 +39,34 @@ const getHeaders = function ({ headers, url }) {
   }
 }
 
+const appendSerializedUrlParams = (url, params) => {
+  if (!params || Object.keys(params).length === 0) {
+    return url
+  }
+  params = stringify(params, { encodeValuesOnly: true, arrayFormat: 'brackets' })
+
+  // Url params may be already appended before passing it to dpApi, this must be handled accordingly.
+  return url.includes('?') ? `${url}&${params}` : `${url}?${params}`
+}
+
 const doRequest = (async ({ url, method = 'GET', data = {}, params, options = {} }) => {
-  const payload = {
-    method,
-    options,
-    params,
-    headers: getHeaders({ ...params, url })
+  const fetchOptions = {
+    headers: getHeaders({ ...params, url }),
+    method
   }
 
   if (method.toUpperCase() !== 'GET') {
     if (data instanceof FormData) {
-      payload.body = data
+      fetchOptions.body = data
```

Comment:
why did You rename payload? at least, I think `fetchOptions` don't clarify what it is. Especially because `doRequest` does more than fetch ;-)

Reply:
I agree that `fetchOptions` is slightly confusing (because it might also be a function name), but `payload` is also not a very descriptive name. We could change it to `settings` or `config`?

Reply:
The variable is passed as the second argument (the "options" argument) into the [`fetch()` global function](https://developer.mozilla.org/en-US/docs/Web/API/fetch#syntax). The decision to name that function "fetch", although it can send all kinds of requests, has been made by some other guys and i only intended to express "these are the options being passed into the fetch function" here.